plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/cache.cpp

```diff
@@ -36,7 +36,7 @@ Cache *copyCache( unsigned char **p, size_t size, Cache *ptr ) {
 
     MK_POINTER( &( ecopy->address ) );
     MK_POINTER( &( ecopy->pointers ) );
-    Hashtable *objectMap = newHashTable( 100 );
+    Hashtable *objectMap = newHashTable( size / 40 );
```

Comment:
this is great - thank you.

two questions:

- why divided by 40, specifically?
- what happens if size is <40? (do we need some check/guard before using `size`)

Reply:
This explanation is great - I definitely appreciate the answer.

Regarding my <40 question, I meant what happens if `size / 40` is less than 1 and therefore goes to zero... (`newHashTable( 0 )`)... Do we need an additional check/guard for that case?

Reply:
When copyCache is called with size < 40, it will start writing memory out of bounds 4 lines earlier:
```
    allocateInBuffer( Cache, ecopy, ptr );
```
since the size of the Cache struct is well over 40. So while I don't think that an *additional* guard is warranted for this case, a *general* one would be useful at the start of the function:
```
    if ( size < DEFAULT_BLOCK_SIZE || size % REGION_ALIGNMENT != 0 ) { /* size should be large enough and divisible by ALIGNMENT */
        return NULL;
    }
```